modules/text2vec-cohere/clients/cohere.go

```diff
@@ -65,20 +81,33 @@ func New(apiKey string, timeout time.Duration, logger logrus.FieldLogger) *vecto
 }
 
 func (v *vectorizer) Vectorize(ctx context.Context, input []string,
-	config ent.VectorizationConfig,
-) (*ent.VectorizationResult, error) {
-	return v.vectorize(ctx, input, config.Model, config.Truncate, config.BaseURL, searchDocument)
+	cfg moduletools.ClassConfig,
+) (*modulecapabilities.VectorizationResult, *modulecapabilities.RateLimits, error) {
+	icheck := ent.NewClassSettings(cfg)
+	config := ent.VectorizationConfig{
+		Model:   icheck.Model(),
+		BaseURL: icheck.BaseURL(),
+	}
+	res, err := v.vectorize(ctx, input, config.Model, config.Truncate, config.BaseURL, searchDocument)
+	return res, nil, err
 }
 
 func (v *vectorizer) VectorizeQuery(ctx context.Context, input []string,
-	config ent.VectorizationConfig,
-) (*ent.VectorizationResult, error) {
-	return v.vectorize(ctx, input, config.Model, config.Truncate, config.BaseURL, searchQuery)
+	cfg moduletools.ClassConfig,
+) (*modulecapabilities.VectorizationResult, *modulecapabilities.RateLimits, error) {
+	icheck := ent.NewClassSettings(cfg)
+	config := ent.VectorizationConfig{
+		Model:    icheck.Model(),
+		BaseURL:  icheck.BaseURL(),
+		Truncate: icheck.Truncate(),
+	}
```

Comment:
make a method out of it `getVectorizationConfig` and use it in `Vectorize` and `VectorizeQuery` methods. Truncate param should be always passed